connect/api/src/main/java/org/apache/kafka/connect/connector/Connector.java

```diff
@@ -121,4 +123,21 @@ public void reconfigure(Map<String, String> props) {
      * Stop this connector.
      */
     public abstract void stop();
+
+    /**
+     * Get the configuration definitions for this connector.
+     * @return ConfigDef the contains all the configuration definition for this connector.
+     */
+    public abstract ConfigDef configuration();
+
+    /**
+     * Validate the connector configuration values against configuration definitions.
+     * @param configDef the configuration definition
+     * @param connectorConfigs the provided configuration values
+     * @return List of Config, each Config contains the updated configuration information given
+     * the current configuration values.
+     */
+    public List<Config> validate(ConfigDef configDef, Map<String, String> connectorConfigs) {
```

Comment:
Why does this take in the `ConfigDef`? Since it is defined by the `Connector` is there any reason to expose it here?

Similarly, do we ever really need the raw `configuration()` or can we just expose something like `public List<Config> validate(Map<String, String> connectorConfigs)`? When we have no config info and just need something to start with, wouldn't calling that with an empty map of configs be sufficient?